--- a/cmake/share/cmake-3.22/Modules/Platform/Android-Initialize.cmake	2021-11-18 23:51:22.000000000 +0800
+++ b/cmake/share/cmake-3.22/Modules/Platform/Android-Initialize.cmake	2021-12-05 10:57:34.498031789 +0800
@@ -77,7 +77,7 @@
 
 # Natively compiling on an Android host doesn't use the NDK cross-compilation
 # tools.
-if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Android")
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Android" AND NOT CMAKE_ANDROID_NDK)
   return()
 endif()
 
